gcloud-java-dns/src/test/java/com/google/gcloud/dns/it/ITDnsTest.java

```diff
@@ -0,0 +1,908 @@
+      .description(ZONE_DESCRIPTION1)
+      .dnsName(ZONE_DNS_NAME_NO_PERIOD)
+      .build();
+  public static final DnsRecord A_RECORD_ZONE1 =
+      DnsRecord.builder("www." + ZONE1.dnsName(), DnsRecord.Type.A)
+          .records(ImmutableList.of("123.123.55.1"))
+          .ttl(25, TimeUnit.SECONDS)
+          .build();
+  public static final DnsRecord AAAA_RECORD_ZONE1 =
+      DnsRecord.builder("www." + ZONE1.dnsName(), DnsRecord.Type.AAAA)
+          .records(ImmutableList.of("ed:ed:12:aa:36:3:3:105"))
+          .ttl(25, TimeUnit.SECONDS)
+          .build();
+  public static final ChangeRequest CHANGE_ADD_ZONE1 = ChangeRequest.builder()
+      .add(A_RECORD_ZONE1)
+      .add(AAAA_RECORD_ZONE1)
+      .build();
+  public static final ChangeRequest CHANGE_DELETE_ZONE1 = ChangeRequest.builder()
+      .delete(A_RECORD_ZONE1)
+      .delete(AAAA_RECORD_ZONE1)
+      .build();
+
+  public static void clear() {
+    Page<Zone> zones = DNS.listZones();
+    Iterator<Zone> zoneIterator = zones.iterateAll();
+    while (zoneIterator.hasNext()) {
+      Zone zone = zoneIterator.next();
+      List<DnsRecord> toDelete = new LinkedList<>();
+      if (zone.name().startsWith(PREFIX)) {
+        Iterator<DnsRecord> dnsRecordIterator = zone.listDnsRecords().iterateAll();
+        while (dnsRecordIterator.hasNext()) {
+          DnsRecord record = dnsRecordIterator.next();
+          if (!ImmutableList.of(DnsRecord.Type.NS, DnsRecord.Type.SOA).contains(record.type())) {
+            toDelete.add(record);
+          }
+        }
+        if (!toDelete.isEmpty()) {
+          ChangeRequest deletion =
+              zone.applyChangeRequest(ChangeRequest.builder().deletions(toDelete).build());
+          waitUntilComplete(zone.name(), deletion.id());
+        }
+        zone.delete();
+      }
+    }
+  }
+
+  private static List<Zone> filter(Iterator<Zone> iterator) {
+    List<Zone> result = new LinkedList<>();
+    while (iterator.hasNext()) {
+      Zone zone = iterator.next();
+      if (zone.name().startsWith(PREFIX)) {
+        result.add(zone);
+      }
+    }
+    return result;
+  }
+
+  @BeforeClass
+  public static void before() {
+    clear();
+  }
+
+  @AfterClass
+  public static void after() {
+    clear();
+  }
+
+  private static void assertEqChangesIgnoreStatus(ChangeRequest expected, ChangeRequest actual) {
+    assertEquals(expected.additions(), actual.additions());
+    assertEquals(expected.deletions(), actual.deletions());
+    assertEquals(expected.id(), actual.id());
+    assertEquals(expected.startTimeMillis(), actual.startTimeMillis());
+  }
+
+  private static void waitUntilComplete(String zoneName, String changeId) {
+    while (true) {
+      ChangeRequest changeRequest = DNS.getChangeRequest(zoneName, changeId,
+          Dns.ChangeRequestOption.fields(Dns.ChangeRequestField.STATUS));
+      if (ChangeRequest.Status.DONE.equals(changeRequest.status())) {
+        return;
+      }
+      try {
+        Thread.sleep(10);
```

Comment:
Nit: is there a reason for checking so frequently? I think we could relax and take a 500ms break